ath79: add support for Trendnet TEW-823DRU

ppancha opened the pull request:
Trendnet TEW-823DRU is a dual-band AC1750 router.
The router section is based on Qualcomm/Atheros QCA9558 + QCA9880.

Specification:

720 MHz CPU
256 MB of RAM
16 MB of FLASH
3T3R 2.4 GHz
3T3R 5 GHz
5x 10/100/1000 Mbps Ethernet

Firmware can be flashed from the web interface with no issues.

Signed-off-by: Pramod Pancha <pancha@vill.com>

adrianschmutzler commented on target/linux/ath79/base-files/etc/board.d/02_network:
```diff
@@ -231,6 +231,10 @@ ath79_setup_interfaces()
 		;;
+	trendnet,tew-823dru)
+	        ucidef_add_switch "switch0" \
+			  "0@eth1" "1:lan:1" "2:lan:2" "3:lan:3" "4:lan:4" "5:wan" "6@eth0"
```
You do not have to specify port numbers twice.
This can be merged with the definition for buffalo,bhr-4grv2

adrianschmutzler commented on target/linux/ath79/base-files/etc/board.d/02_network:
```diff
@@ -343,6 +347,10 @@ ath79_setup_macs()
 		;;
+	trendnet,tew-823dru)
+		lan_mac=$(mtd_get_mac_text "mac" 4)
+		wan_mac=$(mtd_get_mac_text "mac" 24)
```
You can omit the quotes here.

adrianschmutzler commented on target/linux/ath79/base-files/etc/hotplug.d/ieee80211/10_fix_wifi_mac:
```diff
@@ -30,6 +30,14 @@ case "$board" in
+	trendnet,tew-823dru)
+	        # set the 2.4G interface mac address to LAN MAC
+	        [ "$PHYNBR" -eq 1 ] && \
+		        echo $(mtd_get_mac_text mac 4) > /sys${DEVPATH}/macaddress
```
echo $(command) can be replaced with command, so
mtd_get_mac_text mac 4 > /sys${DEVPATH}/macaddress

adrianschmutzler commented on target/linux/ath79/base-files/etc/hotplug.d/ieee80211/10_fix_wifi_mac:
```diff
@@ -30,6 +30,14 @@ case "$board" in
+		        echo $(mtd_get_mac_text mac 4) > /sys${DEVPATH}/macaddress
+	        # set the 5G interface mac address to WAN MAC + 1
+	        [ "$PHYNBR" -eq 0 ] && \
+		        echo $(macaddr_add "$(mtd_get_mac_text mac 24)" 1) > /sys${DEVPATH}/macaddress
```
Same here:
macaddr_add "$(mtd_get_mac_text mac 24)" 1 > /sys${DEVPATH}/macaddress

adrianschmutzler replied:
BTW: Since this is also the case for the other entries in this file, I have just send a patch for that to the list ...

ppancha commented:
@adrianschmutzler Thank you for reviewing the changes. I have updated the commit with the suggested changes and have verified that the changes did not affect firmware functionality on the router.

adrianschmutzler commented on target/linux/ath79/image/generic.mk:
```diff
@@ -664,6 +664,20 @@ define Device/rosinson_wr818
 
+define Device/trendnet_tew-823dru
+  ATH_SOC := qca9558
+  DEVICE_TITLE := Trendnet TEW-823DRU
```
Please update to use DEVICE_VENDOR, DEVICE_MODEL and DEVICE_VARIANT as introduced in:
https://github.com/openwrt/openwrt/commit/8c3dbaf8b49233e275b2dce4e30dc0a2bf8c86c5

adrianschmutzler commented:
I'm currently working on a PR to store the label-mac-address in the DTS (https://github.com/openwrt/openwrt/pull/2159).

I would appreciate if you could provide information about your device:
Please run
`find /proc/device-tree/ -name "*mac-address*"`
on your device (just with the current firmware, no need to update anything; just has to be ath79, no ar71xx).
For each of the returned paths (if there are any), retrieve the mac-address with `cat`, e.g.
`cat /proc/device-tree/ahb/eth@19000000/mac-address`

Now check which of those returned MAC addresses matches the one on the label/sticker/case of your device and tell me the device-tree path leading to it. Note that you do not have to post your MAC addresses here, but just a statement about which match, e.g. like
```
/proc/device-tree/ahb/eth@1a000000/mac-address
/proc/device-tree/ahb/eth@1a000000/mtd-mac-address-increment
/proc/device-tree/ahb/eth@1a000000/mtd-mac-address
/proc/device-tree/ahb/eth@19000000/mac-address <== this one
/proc/device-tree/ahb/eth@19000000/mtd-mac-address
/proc/device-tree/ahb/apb/wmac@18100000/mac-address
/proc/device-tree/ahb/apb/wmac@18100000/mtd-mac-address
```

There may be one, two or no paths giving the correct address. It may also happen that the `find` command returns nothing. Thanks.

adrianschmutzler commented:
Unfortunately, my description is slightly incorrect. The MAC addresses are stored binary, so you can't read with `cat`, but have to use the following code to read from the paths retrieved by `find`:
```
. /lib/functions/system.sh
get_mac_binary "$yourpath" 0
```
The include is obviously only required once. Sorry.

ppancha commented:
The find command does not return any mac-address files under /proc/device-tree/ahb/eth@19000000 for this device. The issue could be that this device stores the mac address in ascii form in the mac partition and there inso way to specify the mac address in the device tree because of this.


> On Jul 5, 2019, at 9:06 AM, Adrian Schmutzler <notifications@github.com> wrote:
> 
> Unfortunately, my description is slightly incorrect. The MAC addresses are stored binary, so you can't read with cat, but have to use the following code to read from the paths retrieved by find:
> 
> . /lib/functions/system.sh
> get_mac_binary "$yourpath" 0
> The include is obviously only required once. Sorry.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/openwrt/openwrt/pull/2117?email_source=notifications&email_token=AKO4GYMQ2JRLZOTXW55QNGLP55BO3A5CNFSM4HXJ2E6KYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODZJPHOY#issuecomment-508752827>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AKO4GYLYBPF5NNVQIDFTW7DP55BO3ANCNFSM4HXJ2E6A>.
>

ppancha committed:
ath79: add support for Trendnet TEW-823DRU

Trendnet TEW-823DRU is a dual-band AC1750 router.
The router is based on Qualcomm/Atheros QCA9558 + QCA9880.

Specification:

720 MHz CPU
256 MB of RAM
16 MB of FLASH
3T3R 2.4 GHz
3T3R 5 GHz
5x 10/100/1000 Mbps Ethernet

Firmware can be flashed from the web interface. Tested on 3 routers
with no issues.

Signed-off-by: Pramod Pancha <pancha@vill.com>

adrianschmutzler commented:
> The issue could be that this device stores the mac address in ascii form in the mac partition and there is ho way to specify the mac address in the device tree because of this.

Thanks for testing.
Indeed, my solution does not work for devices like that. I'm still looking for a good solution to store label
mac-address for them.

ynezz commented:
Thanks! Pulled into my staging tree at https://git.openwrt.org/openwrt/staging/ynezz.git

adrianschmutzler commented:
@ppancha I've now provided an alternate way to provide label MAC address in 02_network. Can you tell which of the MAC addresses from
```
		lan_mac=$(mtd_get_mac_text mac 4)
		wan_mac=$(mtd_get_mac_text mac 24)
```
matches the one on the label?

adrianschmutzler commented on target/linux/ath79/base-files/etc/board.d/02_network:
```diff
@@ -343,6 +344,10 @@ ath79_setup_macs()
+	trendnet,tew-823dru)
+		lan_mac=$(mtd_get_mac_text mac 4)
+		wan_mac=$(mtd_get_mac_text mac 24)
+		;;
```
@ynezz Just found that this can be merged with `dlink,dir-825-c1|\` above, if you still want to change it.
Otherwise, I'll send a patch later.

adrianschmutzler replied:
On a second thought: You might as well leave it. If I receive the label-mac, I will have to split it back anyway.

ppancha replied:
Both MAC addresses are printed on the label but the WAN MAC address has a /FCC designation after it so I assume that maybe the official label MAC address.
